### ISSUE TITLE
Swift format breaks array subscript equality comparison

### ISSUE THREAD

**Issue by Panajev** [nicklockwood/SwiftFormat#390]
Hey @nicklockwood ,

I have just updated swiftformat to v0.40.1 and I have noticed the following transformation by swiftformat that breaks our current code:

From:
```
    func validateImage(_ completion: @escaping (Bool) -> Void) {
        self.getColors { colors in

            guard let compoenents = colors?.background.cgColor.components else {
                completion(false)
                return
            }
            if (compoenents[0] > 0.001 || compoenents[1] > 0.001 || compoenents[2] > 0.001) {
                completion(true)
            } else {
                completion(false)
            }
        }
    }
```

To:
```
    func validateImage(_ completion: @escaping (Bool) -> Void) {
        self.getColors { colors in

            guard let compoenents = colors?.background.cgColor.components else {
                completion(false)
                return
            }
            if (compoenents0.001 || compoenents < [0]0.001 || compoenents < [1][2] > 0.001) {
                completion(true)
            } else {
                completion(false)
            }
        }
    }
```

Right now I am disabling the formatter around this piece of code as I am not sure which rule is applied (I will admit I have not researched the actual rule that was triggered here sorry :/...): 
```
// swiftformat:disable all
    func validateImage(_ completion: @escaping (Bool) -> Void) {
[...]
// swiftformat:enable all
```

Thank you in advance for your help!

Kind Regards,

Goffredo

**Comment by nicklockwood** [comment]
@Panajev this looks like the same issue as https://github.com/nicklockwood/SwiftFormat/issues/388, in which case the fix should be ready shortly.

**Comment by Panajev** [comment]
Thanks @nicklockwood !

**Comment by nicklockwood** [comment]
@Panajev fixed in 0.40.2.
